### PR TITLE
fix: `PolylineLayer` throws exception: "The west longitude can't be smaller than the east longitude"

### PR DESCRIPTION
**Pull request by josxha**
Some linestrings caused the PolylineLayer to throw an assertion exception.
```console
Assertion failed: file:///Users/matthiaskoch/.pub-cache/git/flutter_map-930e8b60e125e630e3c85ab373c0ebb8e9b08d70/lib/src/geo/latlng_bounds.dart:82:16
east >= west
"The west longitude can't be smaller than the east longitude"

packages/flutter_map/src/geo/latlng_bounds.dart 82:28                             unsafe
packages/flutter_map/src/layer/polyline_layer/polyline_layer.dart 171:41          [_aggressivelyCullPolylines]
packages/flutter_map/src/layer/polyline_layer/polyline_layer.dart 138:11          build
```

This bug was reported by borg_1of3 on discord: https://discord.com/channels/951867686378409984/1235305884620689558/1235305884620689558

The bug sneaked into the code base in https://github.com/fleaflet/flutter_map/pull/1834.

### CODE REVIEW

**Review by mootw — APPROVED** [review]
lgtm

**Comment by fmt-Println-MKO** [comment]
thanks for the quick fix :)

**Comment by josxha** [comment]
@mootw thanks for the review and @monsieurtanuki thanks for the comment. I added the requested refactoring, want to take another look before we merge?

**Review by fmt-Println-MKO — APPROVED** [review]

File: lib/src/geo/latlng_bounds.dart [hunk 1]
```diff
@@ -61,21 +73,21 @@ class LatLngBounds {
     required this.south,
     required this.east,
     required this.west,
-  })  : assert(
-            north <= 90, "The north latitude can't be bigger than 90: $north"),
-        assert(north >= -90,
+  })  : assert(north <= maxLatitude,
+            "The north latitude can't be bigger than 90: $north"),
```
Comment (monsieurtanuki) [inline lib/src/geo/latlng_bounds.dart:77]:
```suggestion
            "The north latitude can't be bigger than $maxLatitude: $north"),
```

**Review by fmt-Println-MKO — APPROVED** [review]

File: lib/src/geo/latlng_bounds.dart (hunk 1 above)
Comment (josxha) [reply lib/src/geo/latlng_bounds.dart]: Thanks, you're absolutely right, the const fields should be used here too. 👍

**Review by JaffaKetchup — APPROVED** [review]
LGTM! Should also make it easier for external uses (for example where a `Polygon` is painted over the entire world, and a hole created at the desired location).

**Pull request merged**